Tools/RemoteSynchronization/LightWeightBackendManager.cs

```diff
@@ -0,0 +1,370 @@
+        /// This method retries the operation with a delay if it fails.
+        /// </summary>
+        /// <param name="remotename">The name of the remote file to get.</param>
+        /// <param name="stream">The stream to write the file to.</param>
+        /// <param name="token">A cancellation token to cancel the operation.</param>
+        /// <returns>A task representing the asynchronous get operation.</returns>
+        public Task GetAsync(string remotename, Stream stream, CancellationToken token)
+        {
+            return RetryWithDelay(
+                $"Get {remotename}",
+                async () =>
+                {
+                    await _streamingBackend!.GetAsync(remotename, stream, token).ConfigureAwait(false);
+                    _anyDownloaded = true;
+                },
+                stream,
+                true,
+                token
+            );
+        }
+
+        /// <summary>
+        /// Instantiates the backend if it has not been instantiated yet.
+        /// If the backend is already instantiated, it simply returns.
+        /// If the maximum number of retries has been reached, it throws an InvalidOperationException.
+        /// This method is called internally to ensure that the backend is ready for operations.
+        /// </summary>
+        /// <exception cref="InvalidOperationException">If the maximum number of instantiations has been reached.</exception>
+        private void Instantiate()
+        {
+            if (_backend != null)
+            {
+                // If we already have a backend, we can just return.
+                return;
+            }
+
+            if (_instantiations < _maxRetries)
+            {
+                _backend = Duplicati.Library.DynamicLoader.BackendLoader.GetBackend(_backendUrl, _options);
+                _streamingBackend = _backend as IStreamingBackend;
+                if (_streamingBackend == null)
+                {
+                    _backend.Dispose();
+                    _backend = null;
+                    throw new InvalidOperationException("Backend does not support streaming operations.");
+                }
+                _instantiations++;
+            }
+            else
+            {
+                throw new InvalidOperationException("Maximum number of backend instantiations reached.");
+            }
+        }
+
+        /// <summary>
+        /// Lists the files in the remote backend asynchronously.
+        /// </summary>
+        /// <param name="token">A cancellation token to cancel the operation.</param>
+        /// <returns>An asynchronous enumerable of file entries.</returns>
+        public IAsyncEnumerable<IFileEntry> ListAsync(CancellationToken token)
+        {
+            while (true)
+            {
+                Instantiate();
+
+                try
+                {
+                    return _streamingBackend!.ListAsync(token);
```

Comment:
I realize this is really annoying to get to do retries do to the enumerable. Can we quick fix it? The regular BackendManager flattens the list before returning it, which is not nice, but allows for retries.